src/main/java/uk/gov/hmcts/reform/idam/client/models/UserDetails.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.hmcts.reform.idam.client.models;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.Getter;
+
+import java.util.List;
+import java.util.Optional;
+
+@Getter
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class UserDetails {
+
+    private final String id;
+    private final String email;
+
+    private final String forename;
+    private final String surname;
+    private final List<String> roles;
+
+    public UserDetails(
+        String id,
+        String email,
+        String forename,
+        String surname,
+        List<String> roles
+    ) {
+        this.id = id;
+        this.email = email;
+        this.forename = forename;
+        this.surname = surname;
+        this.roles = roles;
+    }
+
+
+    public Optional<String> getSurname() {
```

Comment:
how does this work with lombok generated getter for the same field?

Reply:
lombok won't generate it, instead this will be used

Reply:
Io forename getter returns `String` and surname getter returns `Optional<String>`.
Is this only here so that you can use streams api in `getFullName`?

Reply:
no the surname field can be blank, full name will join them together if required